src/annotator/frame-observer.js

```diff
@@ -37,22 +46,29 @@ export default class FrameObserver {
     this._mutationObserver.disconnect();
   }
 
+  /**
+   * @param {HTMLIFrameElement} frame
+   */
   _addFrame(frame) {
     if (FrameUtil.isAccessible(frame)) {
       FrameUtil.isDocumentReady(frame, () => {
-        frame.contentWindow.addEventListener('unload', () => {
+        const frameWindow = /** @type {Window} */ (frame.contentWindow);
+        frameWindow.addEventListener('unload', () => {
           this._removeFrame(frame);
         });
         this._handledFrames.push(frame);
-        this._onFrameAdded(frame);
+        this._onFrameAdded?.(frame);
       });
     } else {
       // Could warn here that frame was not cross origin accessible
     }
   }
 
+  /**
+   * @param {HTMLIFrameElement} frame
+   */
   _removeFrame(frame) {
-    this._onFrameRemoved(frame);
+    this._onFrameRemoved?.(frame);
```

Comment:
I have added a third commit that removes the current two step process to start the observation:

1. instantiate the `FrameObserver` constructor
2. `frameObserver.observe(callback1, callback2)`

Instead the callbacks are provided in the constructor and the observation starts immediately.

Reply:
This makes sense to me. I think the reason for the separate `observe` method was an attempt to follow the pattern of various browser `*Observer` APIs (MutationObserver etc.). If so it was an incorrect application of the pattern because in the browser APIs the callback is provided to the constructor and the `observe` method is used to add a target to observe.